src/community/jdbcconfig/src/test/java/org/geoserver/jdbcconfig/config/JDBCGeoServerImplTest.java

```diff
@@ -131,6 +138,44 @@ public void testModifyService() throws Exception {
         assertNull(s3);
     }
 
+    @Test
+    public void testGlobalIsProperlyLocked() throws Exception {
+
+        GeoServerInfo global = geoServer.getFactory().createGlobal();
+        geoServer.setGlobal(global);
+
+        Executors.newSingleThreadExecutor()
+                .execute(
+                        new Runnable() {
+                            @Override
+                            public void run() {
+                                testSupport
+                                        .getFacade()
+                                        .getConfigDatabase()
+                                        .lock(geoServer.getGlobal().getId(), 60000);
+                            }
+                        });
+
+        RunnableFuture<Void> future =
+                new FutureTask<Void>(
+                        new Callable<Void>() {
+                            @Override
+                            public Void call() throws Exception {
+                                geoServer.save(geoServer.getGlobal());
+                                return null;
+                            }
+                        });
+        Executors.newSingleThreadExecutor().execute(future);
+        try {
+            future.get(5, TimeUnit.SECONDS);
+        } catch (TimeoutException ex) {
```

Comment:
Assert.assertThrows would be a cleaner way to check for the exception:
https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThrows(java.lang.Class,%20org.junit.function.ThrowingRunnable)

Reply:
I changed it. Thanks @sikeoka